File: yh_divisor/src/yh_divisor_sub.cpp
#include "ros/ros.h"
#include "yh_divisor/yh_divisor_msg.h"


void msgCallback(const yh_divisor::yh_divisor_msg::ConstPtr& msg)
{
    ROS_INFO("receive msg = %d", msg->stamp.sec);
    ROS_INFO("receive msg = %d", msg->stamp.nsec);
    ROS_INFO("receive msg = %d", msg->data);

    for(int i= 1; i <= msg->data; i++)
    {
        if (msg->data % i == 0 ) 
        {
            printf("%d, ", i);

        }
    }
    printf("\n");
    
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "yh_divisor_sub");
    ros::NodeHandle nh;

    ros::Subscriber sub =  nh.subscribe("yh_divisor_topic", 100, msgCallback);

    ros::spin();  //프로그램 종료 방지 대기

    return 0;

}

File: yh_divisor/scripts/yh_divisor_sub.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

import rospy
from yh_divisor.msg import yh_divisor_msg

def msgCallback(msg):
    rospy.loginfo("receieve message: %d", msg.stamp.secs)
    rospy.loginfo("receieve message: %d", msg.stamp.nsecs)
    rospy.loginfo("receieve message: %d", msg.data)

    
    for i in range(1, msg.data):
        if(msg.data % i == 0):
            print "{0}".format(i),
    print("")


    

def listner():
    rospy.init_node("yh_divisor_sub", anonymous=True)
    rospy.Subscriber("yh_divisor_topic", yh_divisor_msg, msgCallback)

    rospy.spin()

if __name__ == "__main__":
    listner()
